examples/chest.rs

```diff
@@ -0,0 +1,305 @@
+        ServerState {
+            player_list: None,
+            chest: Default::default(),
+            tick: 0,
+        },
+    )
+}
+
+struct Game {
+    player_count: AtomicUsize,
+}
+
+struct ServerState {
+    player_list: Option<PlayerListId>,
+    chest: InventoryId,
+    tick: u32,
+}
+
+#[derive(Default)]
+struct ClientState {
+    entity_id: EntityId,
+    // open_inventory: Option<WindowInventory>,
+}
+
+const MAX_PLAYERS: usize = 10;
+
+const SIZE_X: usize = 100;
+const SIZE_Z: usize = 100;
+
+#[async_trait]
+impl Config for Game {
+    type ServerState = ServerState;
+    type ClientState = ClientState;
+    type EntityState = ();
+    type WorldState = ();
+    type ChunkState = ();
+    type PlayerListState = ();
+
+    fn max_connections(&self) -> usize {
+        // We want status pings to be successful even if the server is full.
+        MAX_PLAYERS + 64
+    }
+
+    fn dimensions(&self) -> Vec<Dimension> {
+        vec![Dimension {
+            fixed_time: Some(6000),
+            ..Dimension::default()
+        }]
+    }
+
+    async fn server_list_ping(
+        &self,
+        _server: &SharedServer<Self>,
+        _remote_addr: SocketAddr,
+        _protocol_version: i32,
+    ) -> ServerListPing {
+        ServerListPing::Respond {
+            online_players: self.player_count.load(Ordering::SeqCst) as i32,
+            max_players: MAX_PLAYERS as i32,
+            player_sample: Default::default(),
+            description: "Hello Valence!".color(Color::AQUA),
+            favicon_png: Some(include_bytes!("../assets/logo-64x64.png").as_slice().into()),
+        }
+    }
+
+    fn init(&self, server: &mut Server<Self>) {
+        let world = server.worlds.insert(DimensionId::default(), ()).1;
+        server.state.player_list = Some(server.player_lists.insert(()).0);
+
+        // initialize chunks
+        for chunk_z in -2..Integer::div_ceil(&(SIZE_Z as i32), &16) + 2 {
+            for chunk_x in -2..Integer::div_ceil(&(SIZE_X as i32), &16) + 2 {
+                world.chunks.insert(
+                    [chunk_x as i32, chunk_z as i32],
+                    UnloadedChunk::default(),
+                    (),
+                );
+            }
+        }
+
+        // initialize blocks in the chunks
+        for chunk_x in 0..Integer::div_ceil(&SIZE_X, &16) {
+            for chunk_z in 0..Integer::div_ceil(&SIZE_Z, &16) {
+                let chunk = world
+                    .chunks
+                    .get_mut((chunk_x as i32, chunk_z as i32))
+                    .unwrap();
+                for x in 0..16 {
+                    for z in 0..16 {
+                        let cell_x = chunk_x * 16 + x;
+                        let cell_z = chunk_z * 16 + z;
+
+                        if cell_x < SIZE_X && cell_z < SIZE_Z {
+                            chunk.set_block_state(x, 63, z, BlockState::GRASS_BLOCK);
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
You can initialize the platform with `chunks.set_block_state` which would be a bit simpler.